src/main/java/org/commcare/formplayer/session/MenuSession.java

```diff
@@ -127,55 +128,51 @@ private void initializeBreadcrumbs() {
     /**
      * Handle a user step, ignoring performance optimizations and not allowing autolaunch actions.
      *
-     * @param input             The user step input
+     * @param input The user step input
      */
     public boolean handleInput(String input) throws CommCareSessionException {
         return handleInput(input, true, false, false);
     }
 
     /**
-     * @param input             The user step input
-     * @param needsDetail       Whether a full entity screen is required for this request
-     *                          or if a list of references is sufficient
-     * @param confirmed         Whether the input has been previously validated,
-     *                          allowing this step to skip validation
-     * @param allowAutoLaunch   If this step is allowed to automatically launch an action,
-     *                          assuming it has an autolaunch action specified.
+     * @param input           The user step input
+     * @param needsDetail     Whether a full entity screen is required for this request
+     *                        or if a list of references is sufficient
+     * @param confirmed       Whether the input has been previously validated,
+     *                        allowing this step to skip validation
+     * @param allowAutoLaunch If this step is allowed to automatically launch an action,
+     *                        assuming it has an autolaunch action specified.
      */
     public boolean handleInput(String input, boolean needsDetail, boolean confirmed, boolean allowAutoLaunch) throws CommCareSessionException {
-        Screen screen = getNextScreen(needsDetail, allowAutoLaunch);
+        Screen screen = getNextScreen(needsDetail);
         log.info("Screen " + screen + " handling input " + input);
-        if(screen == null) {
+        if (screen == null) {
             return false;
         }
         try {
             boolean addBreadcrumb = true;
             if (screen instanceof EntityScreen) {
-                if (input.startsWith("action ") || !confirmed) {
-                    EntityScreen entityScreen = (EntityScreen)screen;
-                    if (input.startsWith("action ") && entityScreen.getAutoLaunchAction() != null) {
-                        sessionWrapper.executeStackOperations(entityScreen.getAutoLaunchAction().getStackOperations(), entityScreen.getEvalContext());
-                        addBreadcrumb = false;
```

Comment:
Removing this looks like it reintroduced the bug fixed in https://github.com/dimagi/formplayer/pull/826/, could you check it out?

Reply:
Thanks for flagging. [Fix here](https://github.com/dimagi/formplayer/pull/843/commits/5d2b69c4c1dd64f0ecbabe1fa91b751c390572c2)

Reply:
sweet, thank you